plugins/outputs/nats/nats_test.go

```diff
@@ -1,46 +1,260 @@
+				fieldsEqualHelper(t, tc.nats.Jetstream.StreamConfig, si.Config, tc.streamConfigCompareFields...)
+			}
+			// Verify that we can successfully write data to the NATS daemon
+			err = tc.nats.Write(testutil.MockMetrics())
+			require.NoError(t, err)
+		})
+	}
+}
+
+func fieldsEqualHelper(t *testing.T, a, b interface{}, fieldNames ...string) {
+	valA := reflect.ValueOf(a)
+	valB := reflect.ValueOf(b)
 
-	servicePort := "4222"
-	container := testutil.Container{
-		Image:        "nats",
-		ExposedPorts: []string{servicePort},
-		WaitingFor:   wait.ForLog("Server is ready"),
+	if valA.Kind() != reflect.Struct || valB.Kind() != reflect.Struct {
+		t.Error("Both parameters must be structs")
+		return
 	}
-	err := container.Start()
-	require.NoError(t, err, "failed to start container")
-	defer container.Terminate()
 
-	server := []string{fmt.Sprintf("nats://%s:%s", container.Address, container.Ports[servicePort])}
-	serializer := &influx.Serializer{}
-	require.NoError(t, serializer.Init())
-	n := &NATS{
-		Servers:    server,
-		Name:       "telegraf",
-		Subject:    "telegraf",
-		serializer: serializer,
+	if valA.Type() != valB.Type() {
+		t.Error("Both parameters must be of the same type")
+		return
 	}
 
-	// Verify that we can connect to the NATS daemon
-	err = n.Connect()
-	require.NoError(t, err)
+	for _, fieldName := range fieldNames {
+		fieldA := valA.FieldByName(fieldName)
+		fieldB := valB.FieldByName(fieldName)
+
+		require.Equal(t, fieldA.Interface(), fieldB.Interface(), "Field %s should be equal", fieldName)
+	}
+}
 
-	// Verify that we can successfully write data to the NATS daemon
-	err = n.Write(testutil.MockMetrics())
-	require.NoError(t, err)
+func Test_extractNestedTable(t *testing.T) {
+	tests := []struct {
+		name    string
+		tomlMap map[string]interface{}
+		keys    []string
+		want    map[string]interface{}
+		wantErr bool
+	}{
+		{
+			name: "Valid Nested Table",
+			tomlMap: map[string]interface{}{
+				"outer": map[string]interface{}{
+					"name": "outer",
+					"inner": map[string]interface{}{
+						"field1": "abc",
+						"field2": "pqr",
+					},
+				},
+			},
+			keys: []string{"outer", "inner"},
+			want: map[string]interface{}{"field1": "abc", "field2": "pqr"},
+		},
+		{
+			name:    "Invalid Key",
+			tomlMap: map[string]interface{}{"key": "value"},
+			keys:    []string{"nonexistent"},
+			wantErr: true,
+		},
+		{
+			name:    "Non-Table Value",
+			tomlMap: map[string]interface{}{"key": "value"},
+			keys:    []string{"key"},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actual, err := extractNestedTable(tt.tomlMap, tt.keys...)
+			if tt.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, tt.want, actual)
+		})
+	}
+}
+
+func TestConfigParsing(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
```

Comment:
This is not an integration test. (No NATS server is being started afaics)
```suggestion
```

Reply:
Thanks for catching